src/forms/subscriptions.ts

```diff
@@ -83,18 +97,20 @@ class FormFields extends BaseFormFields {
     async setState(): Promise<void> {
         // modified node-zendesk to allow hitting triggers/search api
         // returns all triggers for all channels and teams
-
         let search = SubscriptionFields.PrefixTriggersTitle;
         search += SubscriptionFields.RegexTriggerInstance;
         search += this.call.context.mattermost_site_url;
-        const searchReq = this.zdClient.triggers.search(search);
+        const client = this.zdClient as ZDClient;
+        const searchReq = client.triggers.search(search) || '';
         const triggers = await tryPromiseWithMessage(searchReq, 'Failed to fetch triggers');
+        const results: Promise<void>[] = [];
         for (const trigger of triggers) {
-            await this.addChannelTrigger(trigger);
+            results.push(this.addChannelTrigger(trigger));
         }
+        await Promise.all(results);
     }
 
-    async addChannelTrigger(trigger: any): Promise<void> {
+    async addChannelTrigger(trigger: ZDTrigger): Promise<void> {
         const parsedTitle = parseTriggerTitle(trigger.title);
         const channelID = parsedTitle.channelID;
         const channel = await this.mmClient.getChannel(channelID);
```

Comment:
Could we do some caching here during the request, so we don't fetch the same channel more than once?